core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2558,8 +2558,8 @@ public static long getCarbonIndexSize(SegmentFileStore fileStore,
   // Get the total size of carbon data and the total size of carbon index
   public static HashMap<String, Long> getDataSizeAndIndexSize(String tablePath,
       Segment segment) throws IOException {
-    if (segment.getSegmentFileName() != null) {
-      SegmentFileStore fileStore = new SegmentFileStore(tablePath, segment.getSegmentFileName());
+    SegmentFileStore fileStore = new SegmentFileStore(tablePath, segment.getSegmentFileName());
+    if (fileStore.getSegmentFile() != null) {
```

Comment:
same as previous check..Please check if this will fail when segmmentFileName is null